core/src/main/scala/org/ensime/indexer/stringymap/StringyMap.scala

```diff
@@ -95,6 +96,19 @@ package api {
       def toOrientProperty: OrientProperty = OrientProperty(OType.INTEGER)
     }
 
+    implicit object DeclaredAsSPrimitive extends SPrimitive[DeclaredAs] {
+      def toValue(v: DeclaredAs): java.lang.String = if (v == null) null else StringSPrimitive.toValue(v.toString)
+      def fromValue(v: AnyRef): DeclaredAs = StringSPrimitive.fromValue(v) match {
+        case "Method" => DeclaredAs.Method
```

Comment:
might be a fancy shapeless way to do this by creating a `Map` from the instances keyed by their name.